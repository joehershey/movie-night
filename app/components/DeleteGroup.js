import React, { useState } from "react";
import { View, Text, TouchableWithoutFeedback, TextInput } from "react-native";

import { COLORS, STYLES } from "../assets/saved";

function DeleteGroup(props) {
  const [deleteMode, setDeleteMode] = useState(false);

  return (
    <View>
      {!deleteMode && (
        <TouchableWithoutFeedback
          onPress={() => {
            setDeleteMode(true);
          }}
        >
          <View
            style={{
              alignItems: "center",
              alignSelf: "center",
              justifyContent: "center",
              backgroundColor: COLORS.danger,
              borderRadius: 10,
              width: "50%",
              paddingLeft: 10,
              paddingRight: 10,
              margin: 10,
              marginTop: 60,
              height: 80,
            }}
          >
            <Text style={{ color: "white", margin: 5, fontSize: 20 }}>
              Delete Group
            </Text>
          </View>
        </TouchableWithoutFeedback>
      )}
      {deleteMode && (
        <View
          style={{
            width: "80%",
            backgroundColor: "#f2eceb",
            alignSelf: "center",
            borderRadius: 10,
            padding: 10,
            marginTop: 60,
          }}
        >
          <Text style={{ alignSelf: "center", fontWeight: "bold" }}>
            Are you sure?
          </Text>
          <View style={{ flexDirection: "row", justifyContent: "center" }}>
            <TouchableWithoutFeedback
              onPress={() => {
                setDeleteMode(false);
                props.deleteGroupAPI();
              }}
            >
              <View
                style={{
                  alignItems: "center",
                  alignSelf: "center",
                  justifyContent: "center",
                  backgroundColor: COLORS.danger,
                  flex: 1,
                  borderRadius: 10,
                  paddingLeft: 10,
                  paddingRight: 10,
                  margin: 10,
                  height: 60,
                }}
              >
                <Text style={{ color: "white", margin: 5, fontSize: 15 }}>
                  Delete Group
                </Text>
              </View>
            </TouchableWithoutFeedback>
            <TouchableWithoutFeedback
              onPress={() => {
                setDeleteMode(false);
              }}
            >
              <View
                style={{
                  alignItems: "center",
                  alignSelf: "center",
                  justifyContent: "center",
                  backgroundColor: "grey",
                  flex: 1,
                  borderRadius: 10,
                  paddingLeft: 10,
                  paddingRight: 10,
                  margin: 10,
                  height: 60,
                }}
              >
                <Text style={{ color: "white", margin: 5, fontSize: 15 }}>
                  Cancel
                </Text>
              </View>
            </TouchableWithoutFeedback>
          </View>
        </View>
      )}
    </View>
  );
}

export default DeleteGroup;
