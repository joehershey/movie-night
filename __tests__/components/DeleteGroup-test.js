import DeleteGroup from "../../app/components/DeleteGroup";
import React from 'react';
import renderer from 'react-test-renderer';

describe('Test Delete Group', () => {
    beforeEach( async () => {
        global.tree = renderer.create(<DeleteGroup />);
    });

    it("Delete group renders properly", async () => {
        await expect(global.tree).toMatchSnapshot();
    });
});