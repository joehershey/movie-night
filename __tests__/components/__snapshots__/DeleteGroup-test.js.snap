// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Delete Group Delete group renders properly 1`] = `
<View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#b81f00",
        "borderRadius": 10,
        "height": 80,
        "justifyContent": "center",
        "margin": 10,
        "marginTop": 60,
        "paddingLeft": 10,
        "paddingRight": 10,
        "width": "50%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 20,
          "margin": 5,
        }
      }
    >
      Delete Group
    </Text>
  </View>
</View>
`;
